src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = Object.assign({}, state);
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(currentState, action.extraData);
+        result.push({ ...currentState });
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          if (currentState.hasOwnProperty(key)) {
+            delete currentState[key];
+          }
+        }
+        result.push({ ...currentState });
+        break;
+
+      case 'clear':
+        currentState = {};
```

Comment:
that's little hacky I think the point of this task is to modify the `currentState` object, instead of reassigning it to new one - please read: [passing by reference vs value](https://flexiple.com/javascript/javascript-pass-by-reference-or-value/)